noria-server/dataflow/src/ops/filter.rs

```diff
@@ -10,7 +10,7 @@ use prelude::*;
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Filter {
     src: IndexPair,
-    filter: sync::Arc<Vec<Option<FilterCondition>>>,
+    filter: sync::Arc<Vec<(usize, FilterCondition)>>,
```

Comment:
What's the reason for this change? It'll incur some performance overhead if I'm reading the change right?

Reply:
It allows merging ANDed filters over the same columns (e.g. `x < 5 and x > 3`) into a single filter (two entries for the same column index).

Why would it incur performance overhead? Previously, we had to iterate over lots of `None` entries in the vector, and additionally create the columns indexes via `enumerate`, all on the critical path. Seems to me like this should at least be a neutral change from a performance PoV.

Reply:
Previously, given a single input index, you could efficiently determine the corresponding output index. Maybe the inverse lookup is more common though, in which case this is a perf win. I don't know :)

Reply:
I remember changing a lot of places where we iterated over the entire vector. I don't remember whether there are any where we needed to lookup by index.

But yeah, as Malte said, the goal was to make it possible to handle slightly more complex conditions.